### ISSUE TITLE
Potential mismatch between training and evaluation loss

### ISSUE THREAD

**Issue by Dou-Yiming** [dexmal/dexbotic#14]
Hello, 
I have been trying to finetune the CogACT model on Libero data for sanity check with the following command:
`torchrun --nproc_per_node=8 playground/benchmarks/libero/libero_cogact.py`
I've tried multiple different configurations:

- Load the pretrained model from either Dexbotic-Base or Dexbotic-CogACT-SArm
- Freeze/unfreeze the vision module
- Use/Not use LoRA

For each of the model, the training loss will be below 0.02 after several epochs.
However when I load the finetuned checkpoints into the evaluation pipeline, none of them seems to be working (success rate 0%) and the model seems to be always outputting very similar actions regardless of the prompts.

Then I tried directly loading the pretrained libero_cogact checkpoint (which works well in the evaluation pipeline) into the training pipeline, and I found that the initial training loss of this model is around 0.7, which is really high. And if I finetune this model, the training loss will be quickly dropped to around 0.02.

I think this might be suggesting that some of the minor configurations of the training pipeline differs from the evaluation (e.g., the format of the prompts). I noticed that the evaluation prompts start with "A chat between a curious user and an artificial intelligence assistant. ", which might be part of the reason?

Could you double-check if there's any mismatch between training and evaluation?

Thank you in advance!

**Comment by Dou-Yiming** [comment]
Thanks for the quick reply!
I was using exactly the same configuration as libero_cogact.py on 8 L40S GPUs
The training loss goes to around 0.02 after training for 2500+ steps.
However if I load the model into the benchmark it's always resulting in 0% success rate, and the robot seems to be always doing the same action regardless of the prompt (as shown below).
The experiment that confuses me the most is that if I load the libero_cogact checkpoint (which performs perfectly in the evaluation pipeline) into the training pipeline, the loss would be around 0.7, which seems to be very weird.

<img width="1155" height="395" alt="Image" src="https://github.com/user-attachments/assets/a46b607a-5e3c-46cf-8072-7ad4dc49e16b" />

https://github.com/user-attachments/assets/ba67337e-c405-4262-a100-c0b5553080cb

**Comment by yuzhms** [comment]
Thanks for the details. Here are a few notes based on your description:

1. The default configuration in `libero_cogact.py` trains for 65,975 steps. Your run only reached around 2,500 steps, which is far too few. The model is not expected to converge at that point, so the failure in the benchmark is expected.

2. We also tested re-loading the official libero_cogact checkpoint. The initial training loss after loading is typically around 0.02, which matches the final training loss of the pretrained model. Getting 0.7 on your side strongly suggests something inconsistent in the training pipeline.
If you did not modify the original `libero_cogact.py` configuration, the pipelines should be identical. However, there is one common source of mismatch: the dataset used for training and the corresponding normalization statistics.
Please verify the following:
   + Make sure you are training on all four Libero subsets together (goal, spatial, object, and 10). The CogACT checkpoint was pretrained on the combined dataset.
   + Ensure that the normalization statistics you compute match the ones used by the pretrained CogACT model. The official checkpoint includes its normalization stats in `norm_stats.json`. If your computed stats differ, the initial loss will change a lot.
